core/src/main/java/com/linecorp/armeria/common/Flags.java

```diff
@@ -400,6 +400,11 @@ private static boolean validateTransportType(TransportType transportType, String
     private static final boolean TLS_ALLOW_UNSAFE_CIPHERS =
             getValue(FlagsProvider::tlsAllowUnsafeCiphers, "tlsAllowUnsafeCiphers");
 
+    // Maximum 16KiB https://datatracker.ietf.org/doc/html/rfc5246#section-6.2.1
+    private static final int DEFAULT_MAX_CLIENT_HELLO_LENGTH =
+            getValue(FlagsProvider::defaultMaxClientHelloLength, "defaultMaxClientHelloLength",
+                     value -> value >= 0 && value <= 16384);
```

Comment:
Should the valud be? `16777215`?

Reply:
The value is from Netty code. https://github.com/netty/netty/blob/e2c706adad8fa3c5c5d5f15d1b6b2fb0a582a06d/handler/src/main/java/io/netty/handler/ssl/SslClientHelloHandler.java#L45-L45

Reply:
Oops, that's correct. 😓

Reply:
I'm actually looking into the RFC where the value came from. 
https://www.rfc-editor.org/rfc/rfc5246#section-6.2.1

Reply:
Based on the RFC, `16MB` is a bug of Netty?

Reply:
I realized that it could be split on multiple recodes.

https://github.com/netty/netty/commit/535da17e45201ae4278c0479e6162bb4127d4c32
> In theory the ClientHello can span multiple records and so reach the length of 16MB. This can result in high memory usage, we should allow the user to define a maximum length.

Reply:
From my understanding, I think they roughly calculate the value:
```
struct {
    ProtocolVersion client_version; (2B)
    Random random; (32B)
    SessionID session_id; (32B)
    CipherSuite cipher_suites<2..2^16-2>; (64KB)
    CompressionMethod compression_methods<1..2^8-1>; (256B)
    select (extensions_present) {
        case false:
            struct {};
        case true:
            Extension extensions<0..2^16-1>; (64KB)
    };
} ClientHello;
```
Let me investigate it more if I missed something.

Reply:
Our default value is already conservative. If users want to explicitly use a large value, I think we can let users use the value if it does violate specification and Netty allows.

Reply:
It seems like there's no explicit maximum length defined in the RFC:
https://github.com/aws/s2n-tls/issues/4086
However, the struct uses uint24 for the message length so I think Netty used the value:
https://datatracker.ietf.org/doc/html/rfc5246#section-7.4
Let me update the comment.

Reply:
👍 It makes sense.